### PR TITLE
attrs namedtuple

### PR DESCRIPTION
**Pull request by Tinche**
This PR further enhances the attrs plugin by generating a more correct `__attrs_attrs__` field.

The current version of the plugin generates this field as a tuple of attributes. This enhancement makes it a namedtuple, which is what attrs actually does in runtime. The difference is a large increase in the ergonomics of the field: instead of `A.__attrs_attrs__[0]`, a user can access an attribute using `A.__attrs_attrs__.x`. Also note the end-users don't actually use the `__attrs_attrs__` field directly, but access it using `attr.fields`, which is a very simple getter interface.

This might sound like a trivial change, but it has big repercussions down the line for static checks in libraries building on top of attrs.

I have added 2 extra tests and improved an existing one.

### CODE REVIEW

File: mypy/plugins/attrs.py [hunk 1]
```diff
@@ -710,23 +710,38 @@ def _add_init(ctx: 'mypy.plugin.ClassDefContext', attributes: List[Attribute],
...
         ctx.api.named_type_or_none('attr.Attribute', [attr_type or any_type]) or any_type
-        for attr_type in raw_attr_types
+        for _, attr_type in attrs
     ]
     fallback_type = ctx.api.named_type('builtins.tuple', [
         ctx.api.named_type_or_none('attr.Attribute', [any_type]) or any_type,
     ])
-    var = Var(name=attr_name, type=TupleType(attributes_types, fallback=fallback_type))
+
+    namedtuple_cls_name = "_AttrsAttributes"
```
Comment (sobolevn) [inline mypy/plugins/attrs.py:724]: Module-level constant maybe?

Comment (Tinche) [reply mypy/plugins/attrs.py]: Done!

**Comment by Tinche** [comment]
@sobolevn Hello, is there anything more I can do to nudge this along? (I tried the re-request review button, it did nothing.)

**Comment by sobolevn** [comment]
@Tinche Sorry, I am not a real mypy developer, I am just helping with what I can 🙂 
You need to ping someone with proper review permissions.

**Comment by Tinche** [comment]
@euresti Hello, might I kindly ask for your feedback here?

**Comment by euresti** [comment]
@ilevkivskyi, can you give some advice on how to nudge this along?

**Review by ilevkivskyi — APPROVED** [review]
TBH I didn't work on this for a while, but I don't see anything obviously wrong. I would however recommend adding a fine-grained daemon test, where you define a class, then update type of an attribute, and check that the change is propagated correctly to different file in fine-grained mode.

I will leave this for someone else to merge when the fine-grained test is added.

**Comment by Tinche** [comment]
@ilevkivskyi thanks, added a fine-grained test. Hope I got it right!

**Review by JelleZijlstra — APPROVED** [review]

**Pull request merged**